db/collection.go

```diff
@@ -93,16 +93,54 @@ func (c *Collection) Indexes() map[string]Index {
 // Adding an index will override any overlapping index values if they already exist.
 // @note: This does NOT currently build the index. If items have been added prior to adding
 // a new index, they will NOT be indexed a posteriori.
-func (c *Collection) AddIndex(path string, unique bool) error {
-	c.indexes[path] = Index{
+func (c *Collection) AddIndex(config IndexConfig) error {
```

Comment:
All persistence of `IndexConfig` moved in here so that it is persisted no matter if this is called when creating a new Collection from the DB or adding an index on an already created Collection.